bleach/linkifier.py

```diff
@@ -110,9 +110,16 @@ def __init__(self, callbacks=DEFAULT_CALLBACKS, skip_tags=None, parse_email=Fals
         self.url_re = url_re
         self.email_re = email_re
 
-        self.parser = html5lib_shim.HTMLParser(namespaceHTMLElements=False)
+        # Create a parser/tokenizer that allows all HTML tags and escapes
+        # anything not in that list.
+        self.parser = html5lib_shim.BleachHTMLParser(
+            tags=html5lib_shim.HTML_TAGS,
+            strip=False,
+            consume_entities=True,
+            namespaceHTMLElements=False,
+        )
         self.walker = html5lib_shim.getTreeWalker('etree')
-        self.serializer = html5lib_shim.HTMLSerializer(
+        self.serializer = html5lib_shim.BleachHTMLSerializer(
```

Comment:
This is the big change. We switch the Linkifier to use our tokenizer and parser. That accounts for known tags vs. unknown tags and does the right thing in regards to escaping. Because we're using our tokenizer and parser, we need to use our serializer, too, otherwise we get multiple rounds of escaping and linkify isn't idempotent anymore.